src/pd/util.rs

```diff
@@ -279,17 +279,32 @@ where
         let ctx = match ctx {
             Ok(ctx) | Err(ctx) => ctx,
         };
-        let done = ctx.reconnect_count == 0 || ctx.resp.is_some();
+        let done = ctx.reconnect_count == 0
+            || (ctx.resp.is_some() && !Self::should_retry(ctx.resp.as_ref().unwrap()));
         if done {
             Ok(Loop::Break(ctx))
         } else {
             Ok(Loop::Continue(ctx))
         }
     }
 
+    fn should_retry(resp: &Result<Resp>) -> bool {
+        match resp {
+            Ok(_) => false,
+            // all grpc related errors are mapped to Error::Grpc
+            Err(Error::Grpc(err)) => {
+                error!("request failed: {:?}, retry", err);
+                true
+            }
+            // other err are returned by response header from pd, no need to retry
+            Err(_) => false,
```

Comment:
Looks like it cause bootstrap issue when there are 2+ nodes try to bootstrap simultaneously.

Reply:
I check the `bootstrap_cluster()` in `node.rs`, it would return okay when ClusterBootstrapped
```rust
  match self.pd_client.bootstrap_cluster(self.store.clone(), region) {
            Err(PdError::ClusterBootstrapped(_)) => {
                error!("cluster {} is already bootstrapped", self.cluster_id);
                store::clear_prepare_bootstrap(engines, region_id)?;
                Ok(())
            }
            // TODO: should we clean region for other errors too?
            Err(e) => panic!("bootstrap cluster {} err: {:?}", self.cluster_id, e),
            Ok(_) => {
                store::clear_prepare_bootstrap_state(engines)?;
                info!("bootstrap cluster {} ok", self.cluster_id);
                Ok(())
            }
        }
```
And what I change just affects async request, `bootstrap_cluster()` for pd_client use sync request, and it would not retry before.